bundlewrap/repo.py

```diff
@@ -541,88 +541,131 @@ def _build_node_metadata(self, blame=False):
-                            node.name,
-                            metadata_processor_name,
-                        )
-                        if DONE in options:
-                            io.debug(_(
-                                "metadata processor {metaproc} for node {node} "
-                                "has indicated that it need NOT be run again"
-                            ).format(
-                                metaproc=metadata_processor_name,
-                                node=node.name,
-                            ))
-                            blacklisted_metaprocs.add((node_name, metadata_processor_name))
-                            metaproc_returned_DONE = True
-                        else:
-                            io.debug(_(
-                                "metadata processor {metaproc} for node {node} "
-                                "has indicated that it must be run again"
-                            ).format(
-                                metaproc=metadata_processor_name,
-                                node=node.name,
-                            ))
-
-                        blame_defaults = False
-                        if DEFAULTS in options:
-                            processed_dict = merge_dict(
-                                processed_dict,
-                                self._node_metadata_partial[node.name],
-                            )
-                            blame_defaults = True
-                        elif OVERWRITE in options:
-                            processed_dict = merge_dict(
-                                self._node_metadata_partial[node.name],
-                                processed_dict,
-                            )
 
-                        if blame:
-                            blame_changed_paths(
-                                self._node_metadata_partial[node.name],
-                                processed_dict,
-                                node_blame,
-                                "metadata_processor:{}".format(metadata_processor_name),
-                                defaults=blame_defaults,
-                            )
+                node = getattr(metadata_processor, '__node')
+                node_name = node.name
+                node_blame = self._node_metadata_blame[node_name]
+                
+                if (node_name, metadata_processor_name) in blacklisted_metaprocs:
+                    continue
+                
+                # check dependencies
+                dependency_missing = False
+                after = getattr(metadata_processor, '__after')
+                if after:
+                    print('####### {} after {}'.format(metadata_processor_name, after))
+                    for dependency in after:
+                        for name, func in metaprocs.items():
+                            if re.match(name, dependency):
+                                dependency_missing = True
```

Comment:
I don't get it. You're just checking if the dependency is present in the list of metadata processors, but aren't tracking it it has actually run yet?